apis/python/tests/test_experiment_query.py

```diff
@@ -157,8 +158,74 @@ def test_experiment_query_all(soma_experiment):
 
         assert len(ad.layers) == 0
 
+
+@pytest.mark.parametrize("n_obs,n_vars,X_layer_names", [(101, 11, ("data",))])
+def test_experiment_query_to_anndata_no_X(soma_experiment):
+    """When X_name is None, AnnData should have X=None and no layers."""
+    with ExperimentAxisQuery(soma_experiment, "RNA") as query:
+        ad = query.to_anndata(X_name=None)
+        assert ad.X is None
+        assert len(ad.layers) == 0
+        assert ad.n_obs == query.n_obs
+        assert ad.n_vars == query.n_vars
+
+
+def _assert_adata_equivalent(a, b):
+    assert a.n_obs == b.n_obs
+    assert a.n_vars == b.n_vars
+    # X can be None in some cases
+    if a.X is None or b.X is None:
+        assert a.X is None and b.X is None
+    else:
+        assert (a.X != b.X).nnz == 0
+    # layers should be dicts (possibly empty)
+    assert set(a.layers) == set(b.layers)
+    for k in a.layers:
+        assert (a.layers[k] != b.layers[k]).nnz == 0
+    # obs/var equality we dont account for the extra soma_joinid column for a -> ExperimentAxisQuery
+    assert a.obs.label.equals(b.obs.label)
+    assert a.var.label.equals(b.var.label)
+    # obsm/varm keys and shapes
+    assert set(a.obsm) == set(b.obsm)
+    for k in a.obsm:
+        assert a.obsm[k].shape == b.obsm[k].shape
+    assert set(a.varm) == set(b.varm)
+    for k in a.varm:
+        assert a.varm[k].shape == b.varm[k].shape
+    # obsp/varp structural equality
+    assert set(a.obsp) == set(b.obsp)
+    for k in a.obsp:
+        assert (a.obsp[k] != b.obsp[k]).nnz == 0
+    assert set(a.varp) == set(b.varp)
+    for k in a.varp:
+        assert (a.varp[k] != b.varp[k]).nnz == 0
```

Comment:
There is already an `assert_adata_equal` in `apis/python/tests/_util.py` that should be used instead.